### PR TITLE
Avoid polling new tasks when handler pool is full

### PR DESCRIPTION
**Pull request by vitarb**

### CODE REVIEW

File: temporal-sdk/src/main/java/io/temporal/internal/worker/ActivityPollTask.java [hunk 1]
```diff
@@ -75,14 +75,6 @@ public PollActivityTaskQueueResponse poll() {
...
     }
 
-    if (taskQueueActivitiesPerSecond > 0) {
-      pollRequest.setTaskQueueMetadata(
-          TaskQueueMetadata.newBuilder()
-              .setMaxTasksPerSecond(
-                  DoubleValue.newBuilder().setValue(taskQueueActivitiesPerSecond).build())
-              .build());
-    }
-
```
Comment (mfateev) [inline temporal-sdk/src/main/java/io/temporal/internal/worker/ActivityPollTask.java:85]: Why is this removed? Task queue level rate limiting is a very important feature

File: temporal-sdk/src/main/java/io/temporal/internal/worker/PollTaskExecutor.java [hunk 2]
```diff
@@ -52,7 +53,7 @@
     taskExecutor =
         new ThreadPoolExecutor(
             0,
-            options.getTaskExecutorThreadPoolSize(),
+            options.getPollerOptions().getTaskExecutorThreadPoolSize(),
```
Comment (mfateev) [inline temporal-sdk/src/main/java/io/temporal/internal/worker/PollTaskExecutor.java:56]: What is the reason for this change?

File: temporal-sdk/src/main/java/io/temporal/internal/worker/PollerOptions.java [hunk 3]
```diff
@@ -48,6 +48,8 @@ public static PollerOptions getDefaultInstance() {
 
     private int maximumPollRateIntervalMilliseconds = 1000;
 
+    private int taskExecutorThreadPoolSize;
```
Comment (mfateev) [inline temporal-sdk/src/main/java/io/temporal/internal/worker/PollerOptions.java:51]: I don't think poller options is the right place for the executor thread pool size.

File: temporal-sdk/src/main/java/io/temporal/internal/worker/SingleWorkerOptions.java [hunk 4]
```diff
@@ -41,7 +41,6 @@ public static Builder newBuilder(SingleWorkerOptions options) {
     private String identity;
     private String binaryChecksum;
     private DataConverter dataConverter;
-    private int taskExecutorThreadPoolSize = 100;
```
Comment (mfateev) [inline temporal-sdk/src/main/java/io/temporal/internal/worker/SingleWorkerOptions.java:44]: I think it should stay here instead of poller options

File: temporal-sdk/src/main/java/io/temporal/internal/worker/ActivityPollTask.java (hunk 1 above)
Comment (vitarb) [reply temporal-sdk/src/main/java/io/temporal/internal/worker/ActivityPollTask.java:85]: There were two identical blocks, I've assumed that it was a mistake and removed one.

File: temporal-sdk/src/main/java/io/temporal/internal/worker/PollerOptions.java (hunk 3 above)
Comment (vitarb) [reply temporal-sdk/src/main/java/io/temporal/internal/worker/PollerOptions.java:51]: We need to access it in the poller, so it's either this or we should be passing more parameters.

File: temporal-sdk/src/main/java/io/temporal/internal/worker/SingleWorkerOptions.java (hunk 4 above)
Comment (vitarb) [reply temporal-sdk/src/main/java/io/temporal/internal/worker/SingleWorkerOptions.java:44]: Should we pass single worker options to the poller then?

**Comment by vitarb** [comment]
Abandoned in favor of #278
